crates/cairo-addons/src/vm/hint_utils/mod.rs

```diff
@@ -0,0 +1,67 @@
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::hint_utils::get_ptr_from_var_name,
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::{
+        errors::math_errors::MathError,
+        relocatable::{MaybeRelocatable, Relocatable},
+    },
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+
+use std::collections::HashMap;
+pub fn serialize_sequence(
+    name: &str,
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<Vec<Felt252>, HintError> {
+    let ptr = get_ptr_from_var_name(name, vm, ids_data, ap_tracking)?;
+    let len_addr = (ptr + 1)?;
+
+    let len_felt = vm.get_integer(len_addr)?.into_owned();
+    let len =
+        len_felt.try_into().map_err(|_| MathError::Felt252ToUsizeConversion(Box::new(len_felt)))?;
+
+    let data = vm.get_relocatable(ptr)?;
+
+    let mut bytes = Vec::new();
+    for i in 0..len {
+        let byte_addr = (data + i)?;
+        let byte = vm.get_integer(byte_addr)?.into_owned();
+        bytes.push(byte);
+    }
```

Comment:
you could use collect like below in deserialize_sequence 
```rust
(0..len)
        .map(|i| {
            let byte_addr = (data + i)?;
            vm.get_integer(byte_addr).map(|b| b.into_owned())
        })
        .collect::<Result<Vec<_>, _>>()
```

Or if you want to keep the manual push, use `Vec::with_capacity(len)` to avoid reallocating at each push if not mistaken